File: dsa/02-blind-75/arrays-hashing/09-three-sum/solutions/typescript/threeSumBruteForce.ts
export function threeSum(numbers: number[]): number[][] {
  const uniqueTriplets = new Set<string>();

  for (let first = 0; first < numbers.length; first += 1) {
    for (let second = first + 1; second < numbers.length; second += 1) {
      for (let third = second + 1; third < numbers.length; third += 1) {
        if (numbers[first] + numbers[second] + numbers[third] === 0) {
          const triplet = [numbers[first], numbers[second], numbers[third]].sort((a, b) => a - b);
          uniqueTriplets.add(triplet.join(","));
        }
      }
    }
  }

  return [...uniqueTriplets].map((triplet) => triplet.split(",").map(Number));
}

File: dsa/02-blind-75/arrays-hashing/09-three-sum/solutions/python/three_sum_brute_force.py
def three_sum(numbers: list[int]) -> list[list[int]]:
    unique_triplets: set[tuple[int, int, int]] = set()

    for first in range(len(numbers)):
        for second in range(first + 1, len(numbers)):
            for third in range(second + 1, len(numbers)):
                if numbers[first] + numbers[second] + numbers[third] == 0:
                    unique_triplets.add(tuple(sorted((numbers[first], numbers[second], numbers[third]))))

    return [list(triplet) for triplet in unique_triplets]
